src/main/java/org/craftercms/deployer/impl/TargetImpl.java

```diff
@@ -211,11 +211,13 @@ public void run() {
 
         @Override
         public void run() {
+            deploymentLock.lock();
```

Comment:
Locks should be used with try/finally by convention:

lock.lock()
try {
    PROTECTED CODE HERE
} finally {
   lock.unlock()
}